test/integration-tests/shared/deployUniversalRouter.ts

```diff
@@ -1,52 +1,99 @@
+  ZERO_ADDRESS,
   V2_INIT_CODE_HASH_MAINNET,
   V3_INIT_CODE_HASH_MAINNET,
-  ROUTER_REWARDS_DISTRIBUTOR,
-  LOOKSRARE_REWARDS_DISTRIBUTOR,
-  LOOKSRARE_TOKEN,
 } from './constants'
 
 export async function deployRouter(
   permit2: Permit2,
+  weth9Address?: string,
+  seaportAddress?: string,
+  nftxZapAddress?: string,
+  x2y2Address?: string,
+  foundationAddress?: string,
+  sudoswapAddress?: string,
+  nft20ZapAddress?: string,
+  cryptopunksAddress?: string,
+  looksRareAddress?: string,
+  routerRewardsDistributorAddress?: string,
+  looksRareRewardsDistributorAddress?: string,
+  looksRareTokenAddress?: string,
+  v2FactoryAddress?: string,
+  v3FactoryAddress?: string,
   mockLooksRareRewardsDistributor?: string,
-  mockLooksRareToken?: string,
-  mockReentrantProtocol?: string
+  mockLooksRareToken?: string
 ): Promise<UniversalRouter> {
+
+  mockLooksRareRewardsDistributor
+  mockLooksRareToken
   const routerParameters = {
-    permit2: permit2.address,
-    weth9: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
-    seaport: '0x00000000006c3852cbEf3e08E8dF289169EdE581',
-    nftxZap: mockReentrantProtocol ?? '0x0fc584529a2AEfA997697FAfAcbA5831faC0c22d',
-    x2y2: '0x74312363e45DCaBA76c59ec49a7Aa8A65a67EeD3',
-    foundation: '0xcDA72070E455bb31C7690a170224Ce43623d0B6f',
-    sudoswap: '0x2B2e8cDA09bBA9660dCA5cB6233787738Ad68329',
-    nft20Zap: '0xA42f6cADa809Bcf417DeefbdD69C5C5A909249C0',
-    cryptopunks: '0xb47e3cd837dDF8e4c57F05d70Ab865de6e193BBB',
-    looksRare: '0x59728544B08AB483533076417FbBB2fD0B17CE3a',
-    routerRewardsDistributor: ROUTER_REWARDS_DISTRIBUTOR,
-    looksRareRewardsDistributor: mockLooksRareRewardsDistributor ?? LOOKSRARE_REWARDS_DISTRIBUTOR,
-    looksRareToken: mockLooksRareToken ?? LOOKSRARE_TOKEN,
-    v2Factory: V2_FACTORY_MAINNET,
-    v3Factory: V3_FACTORY_MAINNET,
+    permit2: permit2.address || ZERO_ADDRESS,
+    weth9: weth9Address || ZERO_ADDRESS,
+    seaport: seaportAddress || ZERO_ADDRESS,
+    nftxZap: nftxZapAddress || ZERO_ADDRESS,
+    x2y2: x2y2Address || ZERO_ADDRESS,
+    foundation: foundationAddress || ZERO_ADDRESS,
+    sudoswap: sudoswapAddress || ZERO_ADDRESS,
+    nft20Zap: nft20ZapAddress || ZERO_ADDRESS,
+    cryptopunks: cryptopunksAddress || ZERO_ADDRESS,
+    looksRare: looksRareAddress || ZERO_ADDRESS,
+    routerRewardsDistributor: routerRewardsDistributorAddress || ZERO_ADDRESS,
+    looksRareRewardsDistributor: looksRareRewardsDistributorAddress || ZERO_ADDRESS,
+    looksRareToken: looksRareTokenAddress || ZERO_ADDRESS,
+    v2Factory: v2FactoryAddress || ZERO_ADDRESS,
+    v3Factory: v3FactoryAddress || ZERO_ADDRESS,
     pairInitCodeHash: V2_INIT_CODE_HASH_MAINNET,
     poolInitCodeHash: V3_INIT_CODE_HASH_MAINNET,
   }
 
-  const routerFactory = await ethers.getContractFactory('UniversalRouter')
-  const router = (await routerFactory.deploy(routerParameters)) as unknown as UniversalRouter
+  // Initialize the wallet.
+  const provider = Provider.getDefaultProvider()
```

Comment:
Can you create one instance of the provider for all the tests, it will be more convenient I believe, and the same you can do with the deployer. You can check how it's implemented in the `v3-core` repository

Reply:
Can you share the link of example, please?

Reply:
https://github.com/matter-labs/v3-core/blob/v1.0.0-zksync-era/test/shared/zkSyncUtils.ts